tests/test_validation.py

```diff
@@ -1083,3 +1047,692 @@ def test_login(self, dmge: DataModelGraphExplorer) -> None:
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(
+                val_rule_one, Series(["A", "B", "C", "C"])
+            ) == ([], [])
+
+            e, _ = va_obj.cross_validation(
+                val_rule_one, Series(["D"], index=[0], name="PatientID")
+            )
+            assert e
+
+            assert va_obj.cross_validation(val_rule_two, Series([])) == ([], [])
+            assert va_obj.cross_validation(val_rule_two, Series(["A"])) == ([], [])
+            assert va_obj.cross_validation(val_rule_two, Series(["A", "A"])) == ([], [])
+            assert va_obj.cross_validation(val_rule_two, Series(["A", "B"])) == ([], [])
+            assert va_obj.cross_validation(val_rule_two, Series(["A", "B", "C"])) == (
+                [],
+                [],
+            )
+            assert va_obj.cross_validation(
+                val_rule_two, Series(["A", "B", "C", "C"])
+            ) == ([], [])
+
+            e, _ = va_obj.cross_validation(
+                val_rule_two, Series(["D"], index=[0], name="PatientID")
+            )
+            assert e
+
+            assert va_obj.cross_validation(val_rule_three, Series([])) == ([], [])
+            assert va_obj.cross_validation(val_rule_three, Series(["D"])) == ([], [])
+
+            e, _ = va_obj.cross_validation(
+                val_rule_three, Series(["A"], index=[0], name="PatientID")
+            )
+            assert e
+
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={"syn1": cross_val_df1, "syn2": cross_val_df1},
+        ):
+            assert va_obj.cross_validation(val_rule_two, Series([])) == ([], [])
+
+            e, _ = va_obj.cross_validation(
+                val_rule_two, Series(["A"], index=[0], name="PatientID")
+            )
+            assert e
+
+            e, _ = va_obj.cross_validation(
+                val_rule_two, Series(["D"], index=[0], name="PatientID")
+            )
+            assert e
+
+    def test__run_validation_across_target_manifests_failures(
+        self,
+        va_obj: ValidateAttribute,
+        cross_val_df1: DataFrame,
+        cross_val_df3: DataFrame,
+    ) -> None:
+        """Tests for ValidateAttribute._run_validation_across_target_manifests with failures"""
+        with patch.object(
+            schematic.models.validate_attribute.ValidateAttribute,
+            "_get_target_manifest_dataframes",
+            return_value={},
+        ):
+            _, result = va_obj._run_validation_across_target_manifests(
+                project_scope=None,
+                rule_scope="value",
+                access_token="xxx",
+                val_rule="rule comp.att",
+                manifest_col=Series([]),
+                target_column=Series([]),
+            )
+            assert not result
```

Comment:
`result` is indeed a boolean(False) in this particular case (Or it should be).

Reply:
It's a bit unfortunate that the function a return variable that is multiple return types, but it's probably a bit much to split that out due to existing logic. Thoughts?

Reply:
I agree. I find alot of the existing code base full of unfortunate choices :) When I  go into fix things, alot of the challenge is figuring out what the hell is going on, and then trying not to break anything else.

Reply:
I think fixing this is too much for now, it can wait until this whole module gets re-written.

Reply:
> I think fixing this is too much for now, it can wait until this whole module gets re-written.

I agree - what we don't want to do is break something on accident. Hopefully, the effort everyone is making in writing more robust tests can provide insight into what the code is doing now.

We are probably looking at at least 6 months (very likely 9-12) for DCTS.  Something to keep in mind for us to only accept "absolutely necessary" new features.  Thanks again!